pkg/injector/pod_patch.go

```diff
@@ -920,3 +933,15 @@ func getVolumeMounts(pod corev1.Pod) []corev1.VolumeMount {
 
 	return volumeMounts
 }
+
+// isACIVirtualKubelet returns true if the pod is scheduled in ACI.
+// If it is an ACI deployment, the pods have a toleration "azure.com/aci".
+// See https://github.com/virtual-kubelet/azure-aci#schedule-a-pod-in-aci
+func isACIVirtualKubelet(ts []corev1.Toleration) bool {
+	for _, t := range ts {
+		if t.Key == "azure.com/aci" {
+			return true
```

Comment:
I think we need to check for the value too (`&& t.Effect == "NoSchedule"`)

Reply:
Do we need this check? In case they expand the scope of `Effect` later.

Reply:
I think we need it. The Kubernetes docs state:

![image](https://user-images.githubusercontent.com/43508/185208879-ca310a3a-24a5-47e0-937b-2691ac733895.png)

https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/

I actually think we need to check that:

1. `t.Key` is `azure.com/aci`
2. One of:
   1. `t.Operator` is either `Equal` or missing (empty string) **and** `t.Value` is `NoSchedule`
   2. `t.Operator` is `Exist`

Reply:
However this is a bit beyond my understanding of tolerations on K8s. I think the real expert is @yaron2

Reply:
Hello @ItalyPaleAle, in the new Helm mode, the configuration now contains an array of tolerations that must be matched with logical AND. The default configuration is now `[{\"effect\":\"NoSchedule\",\"key\":\"azure.com/aci\"}]`.